cmd/plugin/main.go

```diff
@@ -125,7 +127,7 @@ func main() {
 		server.Start(*endpoint,
 			NewIdentityServer(driverVersion),
 			nil,
-			NewNodeServer(driver, mounter, criClient, secret.CreateStoreOrDie(*icpConf, *icpBin, *nodePluginSA, *enableCache)))
+			NewNodeServer(driver, mounter, criClient, secret.CreateStoreOrDie(*icpConf, *icpBin, *nodePluginSA, *enableCache), *asyncImagePulls))
```

Comment:
nit: I generally dislike modifying arguments for supporting more use-cases, instead has a `ServerOptions` struct where you can define/extend various fields as required in future. It would also keep the caller code path consistent ( specially in tests ).

Reply:
➕ with this suggestion. 

Check out the blog post for a detailed explanation and examples - https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

Reply:
https://github.com/warm-metal/csi-driver-image/issues/76